### PR TITLE
Fix issues with animated module to get section list off the floor.

### PR DESCRIPTION
**Pull request by StephenLPeters**
1) Change the event nodes to properly dereference the config
2) Update the interpolate node to allow for multiple range and output value specifications.
3) Change the names of property set variable to be one character to minimize the risk of exceeding the character limit on composition expressions.   :(

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/2710)

### CODE REVIEW

**Comment by ahimberg** [comment]
>   },

while you're in here, you can add AnimatedExample now too:
{
    key: 'AnimatedExample',
    module: require('react-native/RNTester/js/AnimatedExample'),
  },

---
Refers to: vnext/src/RNTester/RNTesterList.uwp.ts:123 in 0aba894. [](commit_id = 0aba89436b25508dec6404abfcf889153a201944, deletion_comment = False)

File: vnext/src/RNTester/RNTesterList.uwp.ts [hunk 1]
```diff
@@ -73,15 +73,14 @@ const ComponentExamples: Array<IRNTesterExample> = [
     module: require('./KeyboardExtensionExample.uwp'),
   },
   {
-    key: 'ScrollViewSimpleExample',
+    key: 'ScrollViewSimpleExample2',
     module: require('react-native/RNTester/js/ScrollViewSimpleExample')
   },
-  /*  TODO: Enable SectionList when Animation is supported
+  //  TODO: Enable SectionList when Animation is supported
```
Comment (ahimberg) [inline vnext/src/RNTester/RNTesterList.uwp.ts]: > //  TODO: Enable SectionList when Animation is supported [](start = 1, length = 57)

delete comment.  (and what is the 2 for in scrollviewsimplexample above?)

File: vnext/ReactUWP/Modules/Animated/InterpolationAnimatedNode.cpp [hunk 2]
```diff
@@ -78,49 +73,104 @@ namespace react { namespace uwp {
   {
     const auto animation = compositor.CreateExpressionAnimation();
     animation.SetReferenceParameter(s_parentPropsName, parent.PropertySet());
-    animation.SetScalarParameter(s_inputMinName, static_cast<float>(m_inputRange[0]));
-    animation.SetScalarParameter(s_inputMaxName, static_cast<float>(m_inputRange[1]));
-    animation.SetScalarParameter(s_outputMinName, static_cast<float>(m_outputRange[0]));
-    animation.SetScalarParameter(s_outputMaxName, static_cast<float>(m_outputRange[1]));
+    for (uint16_t i = 0; i < m_inputRanges.size(); i++)
```
Comment (ahimberg) [inline vnext/ReactUWP/Modules/Animated/InterpolationAnimatedNode.cpp]: >uint16_t [](start = 9, length = 8)

did this not generate a warning that it should be size_t ?

File: vnext/ReactUWP/Modules/Animated/PropsAnimatedNode.cpp [hunk 3]
```diff
@@ -150,32 +150,32 @@ namespace react { namespace uwp {
...
+        animation.Expression(static_cast<winrt::hstring>(L"ValuePropSet.") + ValueAnimatedNode::s_valueName + L" + ValuePropSet." + ValueAnimatedNode::s_offsetName);
         switch (facadeType)
         {
         case FacadeType::Opacity:
           animation.Target(L"Opacity");
           break;
         case FacadeType::Rotation:
           m_rotationAxis = { 0,0,1 };
-          animation.Expression(L"(ValuePropSet.value + ValuePropSet.offset) * 180 / PI");
+          animation.Expression(static_cast<winrt::hstring>(L"(ValuePropSet.") + ValueAnimatedNode::s_valueName + L" + ValuePropSet." + ValueAnimatedNode::s_offsetName + L") * 180 / PI");
```
Comment (licanhua) [inline vnext/ReactUWP/Modules/Animated/PropsAnimatedNode.cpp]: >static_cast<winrt::hstring>(L"(ValuePropSet.") + ValueAnimatedNode::s_valueName + L" + ValuePropSet." + ValueAnimatedNode::s_offsetName + L") * 180 / PI"); [](start = 31, length = 155)

Can we make it as a predefined variable or constexpr?

File: vnext/ReactUWP/Modules/Animated/NativeAnimatedNodeManager.cpp [hunk 4]
```diff
@@ -230,16 +230,19 @@ namespace react {
       if (m_eventDrivers.count(key))
       {
         auto& drivers = m_eventDrivers.at(key);
+        auto iterator = drivers.begin();
 
-        for (auto iterator = drivers.begin(); iterator != drivers.end(); iterator++)
```
Comment (licanhua) [inline vnext/ReactUWP/Modules/Animated/NativeAnimatedNodeManager.cpp]: I would prefer to create a erase_if template for this kind of problem. [https://en.cppreference.com/w/cpp/experimental/map/erase_if](https://en.cppreference.com/w/cpp/experimental/map/erase_if).
Also it's better to still use for loop like this:
for (auto i = c.begin(), last = c.end(); i != last; ) {
  if (pred(*i)) {
    i = c.erase(i);
  } else {
    ++i;
  }
}

File: vnext/ReactUWP/Modules/Animated/NativeAnimatedNodeManager.cpp [hunk 5]
```diff
@@ -258,7 +261,10 @@ namespace react {
       m_delayedPropsNodes.clear();
       for (const auto tag : delayedPropsNodes)
       {
-        m_propsNodes.at(tag)->StartAnimations();
+        if (m_propsNodes.count(tag))
```
Comment (licanhua) [inline vnext/ReactUWP/Modules/Animated/NativeAnimatedNodeManager.cpp]: >m_propsNodes.count(tag) [](start = 12, length = 23)

prefer (m_propsNodes.count(tag)>0)

File: vnext/ReactUWP/Modules/Animated/FrameAnimationDriver.cpp [hunk 6]
```diff
@@ -39,6 +39,10 @@ namespace react { namespace uwp {
     for (auto frame : m_frames)
     {
       normalizedProgress += step;
+      if (normalizedProgress > 1.0f)
```
Comment (licanhua) [inline vnext/ReactUWP/Modules/Animated/FrameAnimationDriver.cpp]: >normalizedProgress [](start = 10, length = 18)

It's more readable to use std::min(normalizedProgress, 1.0f)

**Pull request merged**